test-app/src/main/java/com/mapbox/navigation/examples/core/MapboxNavigationActivity.kt

```diff
@@ -0,0 +1,437 @@
+                    val point = Point.fromLngLat(rawLocation.longitude, rawLocation.latitude)
+                    val cameraOptions = CameraOptions.Builder()
+                        .center(point)
+                        .zoom(13.0)
+                        .pitch(45.0)
+                        .build()
+                    mapboxMap.easeTo(cameraOptions)
+                    navigationLocationProvider.changePosition(rawLocation)
+                    mapboxNavigation.unregisterLocationObserver(this)
+                }
+
+                override fun onEnhancedLocationChanged(
+                    enhancedLocation: Location,
+                    keyPoints: List<Location>
+                ) {
+                }
+            })
+        }
+        mapboxNavigation.startTripSession()
+    }
+
+    @SuppressLint("MissingPermission")
+    private fun initStyle() {
+        mapboxMap.loadStyleUri(
+            Style.MAPBOX_STREETS,
+            {
+                getGesturePlugin().addOnMapLongClickListener(this)
+            },
+            object : OnMapLoadErrorListener {
+                override fun onMapLoadError(mapViewLoadError: MapLoadError, msg: String) {
+                }
+            }
+        )
+    }
+
+    private fun findRoute(origin: Point, destination: Point) {
+        mapboxNavigation.requestRoutes(
+            RouteOptions.builder()
+                .applyDefaultParams()
+                .accessToken(getMapboxAccessTokenFromResources())
+                .coordinates(listOf(origin, destination))
+                .alternatives(true)
+                .build()
+        )
+    }
+
+    private fun updateCameraToOverview() {
+        viewportDataSource.onRouteChanged(directionsRoute)
+        viewportDataSource.overviewPaddingPropertyOverride(overviewEdgeInsets)
+        viewportDataSource.evaluate()
+        navigationCamera.requestNavigationCameraToOverview()
+    }
+
+    private fun updateCameraToFollowing() {
+        viewportDataSource.followingPaddingPropertyOverride(followingEdgeInsets)
+        viewportDataSource.evaluate()
+        navigationCamera.requestNavigationCameraToFollowing()
+    }
+
+    private fun updateCameraToIdle() {
+        navigationCamera.requestNavigationCameraToIdle()
+    }
+
+    private fun startNavigation() {
+        updateCameraToFollowing()
+        mapboxNavigation.registerLocationObserver(locationObserver)
+        mapboxNavigation.registerRouteProgressObserver(routeProgressObserver)
+        mapboxNavigation.registerBannerInstructionsObserver(bannerInstructionsObserver)
+    }
+
+    private fun stopNavigation() {
+        updateCameraToIdle()
+        clearRouteLine()
+        mapboxNavigation.unregisterLocationObserver(locationObserver)
+        mapboxNavigation.unregisterRouteProgressObserver(routeProgressObserver)
+        mapboxNavigation.unregisterBannerInstructionsObserver(bannerInstructionsObserver)
+    }
+
+    private fun clearRouteLine() {
```

Comment:
This should also clear the arrows.

Reply:
There ain't any Api in RouteArrow to clear it like its being done for the route line.

Reply:
What about `MapboxRouteArrowApi#hideManeuverArrow`?

/cc @cafesilencio

Reply:
That would be a hack. @cafesilencio mentioned he has already introduced clear api in his PR on Route arrow. Once that is available, I can update my code in a new PR. Until then, this should not be a blocker.